packages/react-primitives/src/components/read-only-field/index.tsx

```diff
@@ -0,0 +1,97 @@
+import { Copy } from "../icon/copy";
+import clsx from "clsx";
+import * as styles from "./read-only-field.css";
+import { useEffect, useRef } from "react";
+
+type ReadOnlyPropsField = {
+  value: string;
+  id: string;
+  label?: string;
+  as?: "input" | "textarea" | "div";
+  rows?: number;
+  copy?: boolean;
+  className?: string;
+};
+
+/**
+ *  A component used to display read-only values in forms, with optional right accessory to copy the value to clipboard.
+ */
+export function ReadOnlyField({
+  id,
+  value,
+  label,
+  as = "input",
+  rows,
+  copy,
+  className,
+}: ReadOnlyPropsField) {
+  const Component = as;
+  const inputRef = useRef<HTMLElement>(null);
+
+  const copyToClipBoard = () => {
+    if (typeof window !== "undefined") {
+      navigator.clipboard.writeText(value);
+    }
+  };
+
+  // prevent horizontal scrolling
+  useEffect(() => {
```

Comment:
This looks a bit surprising, what is the problem and can we solve it some other way?

Reply:
The problem here is that the URI is longer than the input. Even if we do text wrap with CSS, I can still scroll horizontally with Magic mouse or touchpad and make the value disappear. I tried 5 different ways of fixing this with CSS and only that JS snipped helped 🤷‍♂️

Reply:
How about `text-overflow: ellipsis;`?

Reply:
Not sure about UX there, in case someone wants to select and copy the code, they wouldn't have the whole string 🤔 I guess any CSS solution would be preferred but if not possible I guess this is OK

Reply:
Right now, two issues are present:
- there is no feedback on Copy icon click (at least some kind of visual feedback should be present indicating that the value has been copied)
- you cannot manually copy the whole code

Reply:
A nice pattern I saw was temporarily switch the icon from the copy icon to checked icon. Also flash the border or something like that